FpsSet.py

```diff
@@ -34,10 +34,20 @@ def __init__(self, ccTLDs, primary, associated_sites=None, service_sites=None):
                                      'primary': primary,
                                      'associatedSites': associated_sites, 
                                      'serviceSites': service_sites}
+    
     def __eq__(self, obj):
       if isinstance(obj, FpsSet) and self.primary == obj.primary:
         if self.ccTLDs == obj.ccTLDs:
           if self.associated_sites == obj.associated_sites:
             if self.service_sites == obj.service_sites:
               return True
-      return False
\ No newline at end of file
+      return False
+    
+    def includes(self, domain, with_ccTLDs=True):
+       if (self.primary == domain or
+                   domain in (self.associated_sites or []) or
+                   domain in (self.service_sites or [])):
+           return True
+       if with_ccTLDs:
+           return domain in (variant for variant in self.ccTLDs.values())
```

Comment:
I don't think this is correct. This generator gives a bunch of lists of domains, but `domain` will never be one of those lists (though it might be *in* one of them). So we need to flatten the lists too.
```suggestion
           return domain in (variant for variant_list in self.ccTLDs.values() for variant in variant_list)
```

Can you add a test case that would have caught this?

Reply:
You're correct, I was misunderstanding the generators. Now fixed, and should be properly tested.